crates/pathfinder/src/sync/checkpoint.rs

```diff
@@ -395,25 +395,43 @@ async fn handle_event_stream(
 /// Takes ownership of transactions and returns them if verification passes.
 async fn check_transactions(
     block: &BlockHeader,
-    transactions: Vec<(Transaction, Receipt)>,
+    transactions: Vec<(TransactionVariant, Receipt)>,
 ) -> anyhow::Result<Option<Vec<(Transaction, Receipt)>>> {
     if transactions.len() != block.transaction_count {
         return Ok(None);
     }
+    let (transaction_variants, receipts): (Vec<TransactionVariant>, Vec<Receipt>) =
+        transactions.into_iter().unzip();
     let transaction_final_hash_type =
         TransactionCommitmentFinalHashType::for_version(&block.starknet_version);
     let (transaction_commitment, transactions) = spawn_blocking({
         move || {
-            let txn_refs = transactions.iter().map(|(t, _)| t).collect::<Vec<_>>();
-            calculate_transaction_commitment(&txn_refs, transaction_final_hash_type)
+            let mut transactions = Vec::new();
+
+            rayon::scope(|s| {
+                s.spawn(|_| {
+                    use rayon::prelude::*;
+
+                    transactions = transaction_variants
+                        .into_par_iter()
+                        .map(|variant| Transaction {
+                            hash: todo!(), //variant.calculate_hash(chain_id, query_only),
```

Comment:
Why is this calculation commented out here?

Reply:
good catch :sweat_smile: , I came to a conclusion that I need to fix this while doing the transaction sync test